recipes/soi/meta.yaml

```diff
@@ -0,0 +1,62 @@
+{% set version = "1.2.0" %}
+{% set sha256 = "d6d1d5fcd5cfcf1b70a641413d8b09190399d5e064fcff515155282adc54f92e" %}
+
+package:
+  name: soi
+  version: {{ version }}
+
+source:
+  url: https://github.com/zhangrengang/SOI/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: ${PYTHON} -m pip install . --no-deps --no-build-isolation --ignore-installed -vv
+  entry_points:
+    - soi = soi.options:main
+  run_exports:
+    - {{ pin_compatible('soi') }}
```

Comment:
please use something like `{{ pin_subpackage("myrecipe", max_pin="x") }}` have a look at the PR description :)